ion-schema/src/isl/mod.rs

```diff
@@ -707,26 +889,25 @@ mod isl_tests {
             ))]);
 
         // construct an ISL model from a schema file
-        let expected_schema_result = schema_system.load_isl_schema(file_name);
-        assert!(expected_schema_result.is_ok());
-        let expected_schema = expected_schema_result.unwrap();
+        let expected_schema = schema_system.load_isl_schema(file_name)?;
 
         // initiate an Ion pretty text writer
-        let mut buffer = Vec::new();
-        let mut writer = TextWriterBuilder::pretty().build(&mut buffer).unwrap();
+        let buffer = Vec::new();
+        let config = WriteConfig::<v1_0::Text>::new(TextFormat::Pretty);
+        let mut writer = Writer::new(config, buffer)?;
 
         // write the previously constructed ISL model into a schema file using `write_to`
-        let write_schema_result = expected_schema.write_to(&mut writer);
+        let write_schema_result = expected_schema.write_as_ion(&mut writer);
```

Comment:
This works, though it can also be written `writer.write(expected_schema)`.

Reply:
It can't be written that way because `IslSchema` does not implement `WriteAsIon`. This is because `WriteAsIon` implementations are allowed to write a single value, but an `IslSchema` is a sequence of top-level values.